velox/common/memory/MemoryPool.h

```diff
@@ -466,6 +481,14 @@ class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
 
   virtual std::string toString() const = 0;
 
+  // Invoked to generate a descriptive memory pool exception error message.
+  // Different from toString(), this method aims to provide more detailed, debug
+  // targeted message. See example implementation from
+  // MemoryPoolImpl::poolFailMessage()
+  virtual std::string poolFailMessage(
```

Comment:
Let's keep the current name capExceedingMessage and this must be called on a root memory pool and requester could either one of its own leaf memory pool or the other leaf pool and explicitly tell this based on the root memory pool check.